### DISCUSSION TITLE
Disk write activity every 15 min

### DISCUSSION THREAD

**Discussion by ChrisFr02** [photoprism/photoprism#1598] [Q&A] [answered]
Hi,
I am using Photoprism (latest version) on Unraid (latest version).
After upgrading to the latest docker build of Photoprism last sunday, I am now facing regular intense writes to my NVME disks every 15 minutes. This has not been the case before the update.
I have already disabled "Webdav Autoindex - Safety Delay" and "Webdav Autoimport- Safety Delay" without any effect.

Is there any timer I may tweak to undo this behaviour?

Best regards
Christian

**Comment by lastzero** [comment, marked as answer]
Might be due to facial recognition. You can disable it or you change the background worker interval - see documented config options in our docs.

**Reply by ChrisFr02** [reply]
Thank You.
Would that be the command "photoprism --wakeup-interval SECONDS" to tweak the background worker interval?

**Reply by lastzero** [reply]
Yes, or the related environment variable.

**Comment by QrchackOfficial** [comment]
I'd just like to confirm that this is still a problem. I'm seeing ~1000 write operations

![image](https://user-images.githubusercontent.com/3584315/201125736-2424a661-acda-4ca5-92f5-861a2d87efbe.png)

as well as I/O full pressure stall, meaning it was waiting for disk:

![image](https://user-images.githubusercontent.com/3584315/201125305-d62b18ed-4ca5-4449-b6d8-96e635a2ddd4.png)

Is there any reason you're still doing 1000 write operations *every 15 minutes*, 24/7, in the background, when the photos have been imported months ago?

Might have something to do with the issue, I'm seeing this in logs:

```
time="2022-11-10T14:49:33Z" level=warning msg="faces: failed removing merged clusters for subject jrjcvbj2hkua2p7k"
time="2022-11-10T15:04:44Z" level=warning msg="faces: failed removing merged clusters for subject jrjcvbm24uvqe65i"

[... (these two lines repeated a bunch of times) ...]

time="2022-11-10T15:04:44Z" level=info msg="faces: merged 88 clusters [11.35434576s]"

[... (15 minutes later the whole thing repeats, ad infinitum) ...]
```

that repeats every 15 minutes.

This was tested on `photoprism/photoprism:latest` which at this time is `Build [221105-7a295cab4]`

**Reply by centralhardware** [reply]
Discussed many times.

**Reply by QrchackOfficial** [reply]
Searching "disk" in both Issues (including closed) and Discussions I get a grand total of 1 threads about this issue - this one. Disk space used by thumbnails of disk read during video playback is expected. Massively overkill disk write when the server is completely idle for no reason is not.

**Reply by piec** [reply]
I've seen this one https://github.com/photoprism/photoprism/issues/1359. Search for PHOTOPRISM_WAKEUP_INTERVAL

**Reply by centralhardware** [reply]
https://github.com/photoprism/photoprism/discussions/2628 for example and there are older issues in exists

**Reply by QrchackOfficial** [reply]
Ah. Why is this not a huge red gotcha warning in the getting started guide right after the install steps by now?

**Reply by centralhardware** [reply]
Contribute to the install guid if you want, it's more architecture peculiarities then error.

**Reply by QrchackOfficial** [reply]
And more importantly, if this is fine to change to a month vs default 15 minutes, then why is this being enabled / set to 15 minutes by default? If "they update certain counters, perform WebDAV sync (if enabled) and auto-import/index", then you either:
- have some crazy disk-intensive counters going on,
- WebDAV with sync is enabled by default for no reason,
- or, auto-import/index doesn't care if there are any new files and goes through all files just in case

All 3 of these are insane reasons tbqh.

**Reply by lastzero** [reply]
Could you take a look at this issue and help us figure out what is going wrong here (you seem to have the same warnings that most users do NOT have)?

- https://github.com/photoprism/photoprism/issues/2806

If you are a sponsor, we can also provide you with personal support to see how you can optimize your installation. And of course you can also contribute.
